llpc/lower/llpcSpirvLowerGlobal.cpp

```diff
@@ -2630,6 +2084,68 @@ void SpirvLowerGlobal::lowerAliasedVal() {
   }
 }
 
+// =====================================================================================================================
+// Lowers task payload.
+void SpirvLowerGlobal::lowerTaskPayload() {
+  GlobalVariable *globalToRemove = nullptr;
+
+  for (GlobalVariable &global : m_module->globals()) {
+    // Skip anything that is not a task payload
+    if (!global.hasMetadata(gSPIRVMD::TaskPayload))
+      continue;
+
+    SmallVector<Constant *, 8> constantUsers;
+
+    for (User *const user : global.users()) {
+      if (Constant *const constantUser = dyn_cast<Constant>(user))
+        constantUsers.push_back(constantUser);
+    }
+
+    for (Constant *const constantUser : constantUsers)
+      replaceConstWithInsts(m_context, constantUser);
+
+    // Record of all the functions that our global is used within.
+    SmallSet<Function *, 4> funcsUsedIn;
+
+    for (User *const user : global.users()) {
+      if (Instruction *const inst = dyn_cast<Instruction>(user))
+        funcsUsedIn.insert(inst->getFunction());
+    }
+
+    for (Function *const func : funcsUsedIn) {
+      m_builder->SetInsertPointPastAllocas(func);
```

Comment:
Should we start to adopt `llvm::convertUsersOfConstantsToInstructions` for these kinds of things?

Reply:
This is a good method. I didn't notice it.